runtime/neumann/src/xcm_config.rs

```diff
@@ -166,11 +166,52 @@ impl ShouldExecute for DenyReserveTransferToRelayChain {
 	}
 }
 
+/// Allows execution from `origin` if it is contained in `T` (i.e. `T::Contains(origin)`) taking
+/// payments into account.
+///
+/// Only allows for sequence `DescendOrigin` -> `WithdrawAsset` -> `BuyExecution`
+pub struct AllowPaidExecWithDescendOriginFrom<T>(PhantomData<T>);
+impl<T: Contains<MultiLocation>> ShouldExecute for AllowPaidExecWithDescendOriginFrom<T> {
+	fn should_execute<RuntimeCall>(
+		origin: &MultiLocation,
+		message: &mut Xcm<RuntimeCall>,
+		max_weight: u64,
+		_weight_credit: &mut u64,
+	) -> Result<(), ()> {
+		log::trace!(
+			target: "xcm::barriers",
+			"AllowPaidExecWithDescendOriginFrom origin: {:?}, message: {:?}, max_weight: {:?}, weight_credit: {:?}",
+			origin, message, max_weight, _weight_credit,
+		);
+		ensure!(T::contains(origin), ());
+		let iter = message.0.iter_mut();
+
+		match iter.take(3).collect::<Vec<_>>().as_mut_slice() {
+			[DescendOrigin(..), WithdrawAsset(..), BuyExecution { weight_limit: Limited(ref mut limit), .. }]
+				if *limit >= max_weight =>
+			{
+				*limit = max_weight;
+				Ok(())
+			},
+
+			[DescendOrigin(..), WithdrawAsset(..), BuyExecution { weight_limit: ref mut limit @ Unlimited, .. }] =>
+			{
+				*limit = Limited(max_weight);
+				Ok(())
+			},
+
+			_ => return Err(()),
+		}
+	}
+}
+
 pub type Barrier = DenyThenTry<
 	DenyReserveTransferToRelayChain,
 	(
 		TakeWeightCredit,
 		AllowTopLevelPaidExecutionFrom<Everything>,
+		// Allow xcm flow where execution fee is paid directly from DescendOrigin account
+		AllowPaidExecWithDescendOriginFrom<Everything>,
```

Comment:
Updates the barrier to allow xcm messages that start with DescendOrigin as described in Astar's docs- https://docs.astar.network/docs/xcm/building-with-xcm/xc-remote-transact#xcm-sequence